app/views/works/_playlist.html.erb

```diff
@@ -0,0 +1,34 @@
+<div class="show-page-audio-playlist-wrapper" data-role="audio-playlist-wrapper">
```

Comment:
We talked about making this a ViewModel, so it can have it's own helpers that go with it, rather than just a template. What do you think?

Reply:
So I gave this a shot, but it ended up giving me more complexity and not less.

Seems to me a `ViewModel` is best suited for a self-contained chunk of HTML, possibly with its own template, and definitely with its own helper methods. The code that splits the members of a work into `audio_members` and `non_audio_members` doesn't pertain to the playlist alone: it affects the whole page, because you want all the rejected `non_audio_members` to show up in the non-playlist part of the page. And you don't want a ViewModel making decisions about stuff that's outside of its purview. So I'm leaving the methods in the decorator for now, where they all make sense together.

Reply:
I mean making the show-page-audio-playlist-wrapper into a ViewModel. The component that shows the player. 

The list of audio children to show could be passed into it. Although I also think it might be fine for it to determine that itself -- it would be deciding what the appropriate things for it to show are, not necessarily making decisions for the page. (As we previously had a version where audio things were repeated at the bottom -- what to show in the playlist, and what to show in other places, are to some extent separate decisions)

Reply:
If they can be separate decisions (and very good point about how the Sufia page *did* repeat some of the members on the page) then that's more manageable, yes. I'm going to give it another shot, then.

Reply:
I think it would be fine to pass into the ViewModel the list of audio Assets to display too. Definitely as a first step. The idea is just that putting it in a ViewModel keeps it more encapsulated, and re-usable somewhere else. ANY helper methods it needs, come with it. (Are there no other helper methods it needs? I confess I'm not looking at the code right now).

Reply:
Yup. There's really only one helper method - figure out whether a member has playable audio or not.

Reply:
i suppose there could become more if we factor logic out of the ERB, or if the complexity of requirements change. I dunno, ViewModels seems good to me for something like this we might want to re-use on another page. Even for one helper method might be enough justification.

Reply:
And if we pass an already-filtered list of assets, that's not even a helper method actually; that can stay in the decorator.

Reply:
This ended up working nicely on the second try.